shared/src/types/ethereum_events.rs

```diff
@@ -33,20 +66,149 @@ pub struct TransferToNamada {
+/// fraction that is between zero and one inclusive.
+#[derive(Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+pub struct FractionalVotingPower(Ratio<u64>);
+
+impl FractionalVotingPower {
+    /// Create a new FractionalVotingPower. It must be between zero and one
+    /// inclusive.
+    pub fn new(numer: u64, denom: u64) -> Result<Self> {
+        if denom == 0 {
+            return Err(eyre!("denominator can't be zero"));
+        }
+        let ratio: Ratio<u64> = (numer, denom).into();
+        if ratio > 1.into() {
+            return Err(eyre!(
+                "fractional voting power cannot be greater than one"
+            ));
+        }
+        Ok(Self(ratio))
+    }
+}
+
+impl From<&FractionalVotingPower> for (u64, u64) {
+    fn from(ratio: &FractionalVotingPower) -> Self {
+        (ratio.0.numer().to_owned(), ratio.0.denom().to_owned())
+    }
+}
+
+impl BorshSerialize for FractionalVotingPower {
+    fn serialize<W: ark_serialize::Write>(
+        &self,
+        writer: &mut W,
+    ) -> std::io::Result<()> {
+        let (numer, denom): (u64, u64) =
+            TryFrom::<&FractionalVotingPower>::try_from(&self).map_err(
+                |err| {
+                    std::io::Error::new(
+                        std::io::ErrorKind::InvalidData,
+                        format!(
+                            "Could not serialize {:?} to Borsh: {:?}",
+                            self, err
+                        ),
+                    )
+                },
+            )?;
+        (numer, denom).serialize(writer)
+    }
+}
+
+impl BorshDeserialize for FractionalVotingPower {
+    fn deserialize(buf: &mut &[u8]) -> std::io::Result<Self> {
+        let (numer, denom): (u64, u64) = BorshDeserialize::deserialize(buf)?;
+        Ok(FractionalVotingPower(Ratio::<u64>::new(numer, denom)))
+    }
+}
+
+impl BorshSchema for FractionalVotingPower {
+    fn add_definitions_recursively(
+        definitions: &mut std::collections::HashMap<
+            borsh::schema::Declaration,
+            borsh::schema::Definition,
+        >,
+    ) {
+        let fields =
+            borsh::schema::Fields::UnnamedFields(borsh::maybestd::vec![
+                u64::declaration(),
+                u64::declaration()
+            ]);
+        let definition = borsh::schema::Definition::Struct { fields };
+        Self::add_definition(Self::declaration(), definition, definitions);
+    }
+
+    fn declaration() -> borsh::schema::Declaration {
+        "FractionalVotingPower".into()
+    }
+}
+
 /// This is created by the block proposer based on the Ethereum events included
 /// in the vote extensions of the previous Tendermint round
 #[derive(Debug, Clone, BorshSerialize, BorshDeserialize, BorshSchema)]
 pub struct MultiSignedEthEvent {
     /// Address and voting power of the signing validators
-    pub signers: Vec<(Address, u64)>,
+    pub signers: Vec<(Address, FractionalVotingPower)>,
```

Comment:
Re. discussion before, I still think it should be `EstablishedAddress` rather than `Address`, as only the `Address::Established` variant will make sense here (validators should all be established accounts). Although we may only handle these addresses opaquely, we can encode in the type what we expect the type of addresses to be